Question1. Create a function that takes three arguments a, b, c and returns the sum of the
numbers that are evenly divided by c from the range a, b inclusive.
Examples
evenly_divisible(1, 10, 20) ➞ 0
# No number between 1 and 10 can be evenly divided by 20.
evenly_divisible(1, 10, 2) ➞ 30
# 2 + 4 + 6 + 8 + 10 = 30
evenly_divisible(1, 10, 3) ➞ 18
# 3 + 6 + 9 = 18

In [5]:
def evenly_divisible(a,b,c):
    sum=0
    for i in range(a,b+1):
        if(i%c==0):
            print(i,end = ',')
            sum=sum+i
    return sum
            

In [6]:
evenly_divisible(12,1111,12)

12,24,36,48,60,72,84,96,108,120,132,144,156,168,180,192,204,216,228,240,252,264,276,288,300,312,324,336,348,360,372,384,396,408,420,432,444,456,468,480,492,504,516,528,540,552,564,576,588,600,612,624,636,648,660,672,684,696,708,720,732,744,756,768,780,792,804,816,828,840,852,864,876,888,900,912,924,936,948,960,972,984,996,1008,1020,1032,1044,1056,1068,1080,1092,1104,

51336

Question2. Create a function that returns True if a given inequality expression is correct and
False otherwise.
Examples
correct_signs(&quot;3 &lt; 7 &lt; 11&quot;) ➞ True
correct_signs(&quot;13 &gt; 44 &gt; 33 &gt; 1&quot;) ➞ False
correct_signs(&quot;1 &lt; 2 &lt; 6 &lt; 9 &gt; 3&quot;) ➞ True

In [11]:
eval('3 < 7 > 11')

False

In [12]:
def correct_sign (a):
    if (eval(a)):
        print(True)
    else:
        print(False)

In [13]:
correct_sign('3 < 7 > 11')

False


In [14]:
correct_sign('3 < 7 <11')

True


Question3. Create a function that replaces all the vowels in a string with a specified character.
Examples
replace_vowels(&quot;the aardvark&quot;, &quot;#&quot;) ➞ &quot;th# ##rdv#rk&quot;
replace_vowels(&quot;minnie mouse&quot;, &quot;?&quot;) ➞ &quot;m?nn?? m??s?&quot;
replace_vowels(&quot;shakespeare&quot;, &quot;*&quot;) ➞ &quot;sh*k*sp**r*&quot;

In [20]:
def replace_vowels(a,b):
    b=''
    for i in range(len(a)):
        if a[i] in ['a','e','i','o','u']:
            b=b+'#'
        else:
            b=b+a[i]
    return b

In [21]:
replace_vowels('dsjfjfsoajhcelulmhexmlx','#')

'dsjfjfs##jhc#l#lmh#xmlx'

Question4. Write a function that calculates the factorial of a number recursively.
Examples
factorial(5) ➞ 120
factorial(3) ➞ 6
factorial(1) ➞ 1
factorial(0) ➞ 1

In [22]:
def factorial (n):
    fac=1
    for i in range(1,n+1):
        fac=fac*i
    return fac
        

In [28]:
factorial(5)

120

Question 5
Hamming distance is the number of characters that differ between two strings.
To illustrate:
String1: &quot;abcbba&quot;
String2: &quot;abcbda&quot;
Hamming Distance: 1 - &quot;b&quot; vs. &quot;d&quot; is the only difference.
Create a function that computes the hamming distance between two strings.
Examples
hamming_distance(&quot;abcde&quot;, &quot;bcdef&quot;) ➞ 5
hamming_distance(&quot;abcde&quot;, &quot;abcde&quot;) ➞ 0
hamming_distance(&quot;strong&quot;, &quot;strung&quot;) ➞ 1

In [30]:
def hamming_distance(a,b):
    count=0
    if len(a)==len(b):
        for i in range(len(a)):
            if(a[i]!=b[i]):
                count=count+1
    return count
            

In [31]:
hamming_distance('rahul','bajaj')

4